### DISCUSSION TITLE
[HELP] Polybar custom modules stop responding after initial left-click interaction in AwesomeWM

### DISCUSSION THREAD

**Discussion by SoulThy** [polybar/polybar#2967] [Q&A] [answered]
When Polybar starts up with AwesomeWM, custom modules that work on left-click function correctly for the first time. However, after the initial click, the modules stop responding. This issue seems to be related to AwesomeWM treating Polybar as a regular window.

**My Custom Module Configuration:**
```
  [module/lock]
  type = custom/script
  exec = echo "⏼"
  click-right = /usr/bin/lock.sh
```
  
**Why I Believe this is an AwesomeWM Issue:**

This problem occurs even when I left-click on an empty space in the Polybar and then attempt to click the custom module. It suggests that the issue is not specific to the custom module itself, but rather how AwesomeWM interacts with Polybar.

**What I've Tried (but didn't work):**
I attempted to address the issue by adding the following rule to my AwesomeWM configuration:
```
awful.rules.rules = {
  { rule = { class = "Polybar"  },
     properties = { skip_taskbar = true  } },
     ...
     ...
```
However, even after adding this rule, the problem persists.

**Comment by patrick96** [comment]
We've had this issue before and I also believe the main problem is on the AwesomeWM side (polybar doesn't even receive the mouse events). There is a small section on it on our [known issues page](https://github.com/polybar/polybar/wiki/Known-Issues#mouse-interactions-not-working-in-awesomewm).

Basically, use `override-redirect = true` in your polybar config to tell awesomeWM to not manage polybar, thus not treating it as a regular window.

**Comment by SoulThy** [comment]
Thank you. I must admit, I feel a bit foolish for not checking the known issues page beforehand. Regarding the problem with `override-redirect = true`, I noticed that when I fullscreen applications like YouTube, the polybar remains on top. I've considered a potential solution, which involves running a script that detects when something is in fullscreen mode and deactivates the polybar accordingly. However, at the moment, I don't think I'll pursue this option, it feels like too much work. Instead, I've reconfigured my buttons to be activated only through right-click, and I'll make an effort to avoid left-clicking on the polybar. :)

**Reply by patrick96** [reply]
The solution for this is the `wm-restack` option. There is no awesomewm specific setting, but `generic` may work. If you are building polybar from `master`, `generic` should work even better.

**Comment by SoulThy** [comment]
Unfortunately, I have tested different configurations, including setting `wm-restack` to generic, `bspwm`, and `i3`, but none of them produced the desired results.

- `generic` : the Polybar ends up hidden behind windows every window. 
-  `bspwm` and `i3` : the Polybar remains on top of other windows because of the ` override-redirect `setting.

Ideally, I would like the Polybar to consistently remain at the top of the screen, without being treated as a movable window by awesomeWM. Additionally, it should hide when entering fullscreen mode to avoid obstructing the content.

**Reply by patrick96** [reply]
So what you are trying to achieve is that polybar is in front of windows, and go behind them when they're in fullscreen mode?

I'm not sure this is possible on the polybar side. Polybar would have to somehow detect when any window goes in or out of fullscreen mode and position itself appropriately.

May I ask why you want polybar to be on top of other windows (and thus obstruct them) instead of having a gap between the windows and the edge of the screen (top or bottom) where polybar can live?

**Comment by SoulThy** [comment, marked as answer]
### My Solution
 I managed to fix this issue by modifying the `awesome/rc.lua` (_awesomeWM config file_). Previously, the code in the file was:
```lua
Clientbuttons = gears.table.join(
    awful.button({}, 1, function(c)
        c:emit_signal("request::activate", "mouse_click", { raise = true })
    end),
    awful.button({ modkey }, 1, function(c)
        c:emit_signal("request::activate", "mouse_click", { raise = true })
        awful.mouse.client.move(c)
    end),
    awful.button({ modkey }, 3, function(c)
        c:emit_signal("request::activate", "mouse_click", { raise = true })
        awful.mouse.client.resize(c)
    end)
)
```
I made the following modifications to the code:
```lua
Clientbuttons = gears.table.join(
    -- awful.button({}, 1, function(c)
    --     c:emit_signal("request::activate", "mouse_click", { raise = true })
    -- end),
    awful.button({ modkey }, 1, function(c)
        c:emit_signal("request::activate", "mouse_click", { raise = true })
        awful.mouse.client.move(c)
    end),
    awful.button({ modkey }, 3, function(c)
        c:emit_signal("request::activate", "mouse_click", { raise = true })
        awful.mouse.client.resize(c)
    end)
)
```
---
The first button binding was commented out by wrapping it with` -- `. This means it will no longer be active. The rest of the code remains unchanged.

**Reply by juanibiapina** [reply]
this deactivates focusing on any window using mouse clicks, right?
